sw/host/opentitanlib/src/otp/mod.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod lc_state;
+pub mod otp;
+pub mod otp_img;
+pub mod otp_mmap;
+pub mod vmem_serialize;
+
+mod num_de;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use anyhow::Result;
+    use std::path::Path;
+
+    #[test]
+    fn test_vmem_serialize() -> Result<()> {
+        let mut otp_mmap = otp_mmap::OtpMap::new(Path::new("tests/otp_ctrl_mmap.hjson"))?;
+        let mut otp_img = otp_img::OtpImg::new(Path::new("tests/otp_ctrl_img_dev.hjson"))?;
+        let lc_state = lc_state::LcSecded::new(Path::new("tests/lc_ctrl_state.hjson"))?;
+        let vmem = otp_mmap.make_vmem(&mut otp_img)?;
+        let keys = otp_mmap.generate_keys(&otp_img);
+        let result = vmem.generate(keys, &lc_state)?;
+        let expected = std::fs::read_to_string(Path::new("tests/test.vmem"))?;
+        let expected = expected.split("\n").collect::<Vec<&str>>();
+        println!("{}", result.join("\n"));
+
+        for i in 0..result.len() {
+            assert_eq!(result[i], expected[i]);
```

Comment:
Would comparing the values directly make more sense?

Reply:
I went with your suggestion here, but the reason I was doing this is because both `result` and `expected` are massive, and if they don't equal the console output is a total mess. If there's a more idiomatic way of accomplishing a line by line comparison that may be cleaner.

Reply:
Perhaps we should pull in https://docs.rs/pretty_assertions/latest/pretty_assertions/index.html? @alphan